### ISSUE TITLE
[Known Issue]: The `thinking block` is not guaranteed for showing 'thinking' as expected for Claude Haiku.

### ISSUE THREAD

**Issue by lumpinif** [richards199999/Thinking-Claude#45]
The `thinking code block` does not initially display or render for the user when using Claude Haiku. However, it can reappear if you prompt Claude to explicitly express the thinking within the code block.

This unknown reason may be affecting all prompts right now.

![image](https://github.com/user-attachments/assets/34fea3e2-25ac-4d7c-82be-211253547606)

**Comment by lumpinif** [comment]
## [Possible Solution]
replacing the first guideline with 
“- Claude MUST always ENSURE every Claude's thinking process MUST be expressed in `code blocks` with 'thinking' header even including the fisrt response.”

**Comment by richards199999** [comment]
yes this is indeed a issue. Haiku's instruction-following ability is not good enough...

**Comment by lumpinif** [comment]
> yes this is indeed a issue. Haiku's instruction-following ability is not good enough...

yes just in case anybody only has access to Haiku for now
